pancancer_classifier.py

```diff
@@ -134,17 +154,37 @@ def get_threshold_metrics(y_true, y_pred, tissue='all'):
     return {'auroc': auroc, 'roc_df': roc_df, 'tissue': tissue}
 
 # Load Data
-rnaseq_df = pd.read_table('data/rnaseq_data.tsv', index_col=0)
-mutation_df = pd.read_table('data/mutation_table.tsv', index_col=0)
-clinical_df = pd.read_table('data/clinical_data.tsv', index_col=0,
-                            low_memory=False)
+if xena:
+    expr_fh = 'data/xena/expression.tsv.bz2'
+    mut_fh = 'data/xena/xena_mutation_table.tsv'
+    clin_fh = 'data/xena/samples.tsv'
+    gene_map = pd.read_table('data/xena/HiSeqV2-gene-map.tsv', index_col=0)
+    tcga_acronym = pd.read_table('data/tcga_dictionary.tsv', index_col=0)
+    ac = tcga_acronym.to_dict()['acronym']
+else:
+    expr_fh = 'data/rnaseq_data.tsv'
+    mut_fh = 'data/mutation_table.tsv'
+    clin_fh = 'data/clinical_data.tsv'
+
+rnaseq_df = pd.read_table(expr_fh, index_col=0)
+mutation_df = pd.read_table(mut_fh, index_col=0)
+clinical_df = pd.read_table(clin_fh, index_col=0, low_memory=False)
 
 # Subset data
-rnaseq_df.drop('SLC35E2', axis=0, inplace=True)
+if xena:
+    gene_map_dict = dict((str(v), str(k)) for k, v in
+                         gene_map.to_dict()['entrez_gene_id'].items())
+    rnaseq_df = rnaseq_df.rename(index=str, columns=gene_map_dict).T
+    clinical_df = clinical_df.assign(acronym=clinical_df
+                                     .disease.replace(to_replace=ac))
+    base_add = 'XENA'
+else:
+    rnaseq_df.drop('SLC35E2', axis=0, inplace=True)
+    base_add = 'synapse'
 
 # Generate file names for output
-base_fh = 'tissues_' + args.tissues.replace(',', '_') + '_genes_' + \
-          args.genes.replace(',', '_')
+base_fh = base_add + '_tissues_' + args.tissues.replace(',', '_') + \
```

Comment:
I like this approach:

```python
'{}_tissues_{}_genes_{}'.format(
    base_add,
    args.tissues.replace(',', '_'),
    args.genes.replace(',', '_')
}
```

While not much simpler ATM, it is good for your training with Python 3.6 around the corner and the ability to use the advanced [Format Specification Mini-Language](https://docs.python.org/3.4/library/string.html#format-specification-mini-language).